ui/components/app/modals/export-private-key-modal/export-private-key-modal.component.js

```diff
@@ -14,102 +23,123 @@ import {
+  const [showWarning, setShowWarning] = useState(true);
+  const [showHoldToReveal, setShowHoldToReveal] = useState(false);
+  const trackEvent = useContext(MetaMetricsContext);
+  const t = useI18nContext();
 
-  exportAccountAndGetPrivateKey = (password, address) => {
-    const { exportAccount } = this.props;
+  useEffect(() => {
+    return () => {
+      clearAccountDetails();
+      hideWarning();
+    };
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
 
-    exportAccount(password, address)
-      .then((privateKey) => {
-        this.context.trackEvent({
+  const exportAccountAndGetPrivateKey = async (passwordInput, address) => {
+    try {
+      const privateKeyRetrieved = await exportAccount(passwordInput, address);
+      trackEvent(
+        {
           category: MetaMetricsEventCategory.Keys,
           event: MetaMetricsEventName.KeyExportRevealed,
           properties: {
             key_type: MetaMetricsEventKeyType.Pkey,
           },
-        });
-
-        this.setState({
-          privateKey,
-          showWarning: false,
-        });
-      })
-      .catch((e) => {
-        this.context.trackEvent({
+        },
+        {},
+      );
+      setPrivateKey(privateKeyRetrieved);
+      setShowWarning(false);
+      setShowHoldToReveal(true);
+    } catch (e) {
+      trackEvent(
+        {
           category: MetaMetricsEventCategory.Keys,
           event: MetaMetricsEventName.KeyExportFailed,
           properties: {
             key_type: MetaMetricsEventKeyType.Pkey,
             reason: 'incorrect_password',
           },
-        });
+        },
+        {},
+      );
 
-        log.error(e);
-      });
+      log.error(e);
+    }
   };
 
-  renderPasswordLabel(privateKey) {
+  const renderPasswordLabel = (privateKeyInput) => {
     return (
-      <span className="export-private-key-modal__password-label">
-        {privateKey
-          ? this.context.t('copyPrivateKey')
-          : this.context.t('typePassword')}
-      </span>
+      <Text
+        as="span"
+        color={Color.textDefault}
+        marginBottom={2}
+        variant={TextVariant.bodySm}
+      >
+        {privateKeyInput ? t('copyPrivateKey') : t('typePassword')}
+      </Text>
     );
-  }
-
-  renderPasswordInput(privateKey) {
-    const plainKey = privateKey && stripHexPrefix(privateKey);
+  };
 
-    if (!privateKey) {
+  const renderPasswordInput = (privateKeyInput) => {
+    if (!privateKeyInput) {
       return (
-        <input
-          type="password"
+        <TextField
+          width={BLOCK_SIZES.FULL}
+          placeholder={t('enterPassword')}
+          type={TEXT_FIELD_TYPES.PASSWORD}
           className="export-private-key-modal__password-input"
+          onChange={(event) => setPassword(event.target.value)}
           data-testid="password-input"
-          onChange={(event) => this.setState({ password: event.target.value })}
         />
       );
     }
 
```

Comment:
When we refactor these older components, we want to aim to do away with all these renderX methods, because they are often smells that there needs to be a new component or are unnecessary. For example this one is rendered just once, it'd be better to inject this TextField into the primary JSX tree of the component

Reply:
sure let me refactor this